src/libraries/System.Net.Http/src/System/Net/Http/Headers/GenericHeaderParser.cs

```diff
@@ -114,7 +114,7 @@ private static int ParseMultipleEntityTags(string value, int startIndex, out obj
             return resultLength;
         }
 
-        private static int ParseMailAddress(string value, int startIndex, out object? parsedValue)
+        private static int ParseFrom(string value, int startIndex, out object? parsedValue)
```

Comment:
We could consider changing the name of this method (and the related Parser above) since it's not specific to From.

But it's not obvious to me what a better name is. "ParseUnvalidated"? "ParseWithoutValidation"?

I don't think it's worth the effort.

Maybe just add a comment that says "Allow arbitrary header value for From header (aside from newline, which is always invalid in a header value)"

Reply:
Done. Thanks!